### DISCUSSION TITLE
Where to listen for events in streams?

### DISCUSSION THREAD

**Discussion by Jakobud** [sindresorhus/got#102] [Q&A] [answered]
Sorry this is more of a question than a bug report. When doing a streamed `PUT`, where do I listen for the `request`, `response`, `error` etc events?

``` js
fs.createReadStream(fileInput).pipe(got.stream.put(url, options, function(err, data, res)).on('response', function(res){
    console.log(res);
}));
```

That doesn't work. Neither does this:

``` js
fs.createReadStream(fileInput).pipe(got.stream.put(url, options, function(err, data, res){
    this.on('response', function(response){
        console.log(response);
    });
}));
```

Do you have any examples of how to do this?

**Comment by floatdrop** [comment, marked as answer]
@Jakobud afaik you can't mix callback mode with stream mode.

``` js
fs.createReadStream(fileInput)
    .pipe(got.stream.put(url, options))
    .on('response', (res) => { });
    .pipe(process.stdout);
```

---

Such questions are better asked at [Stack Overflow](http://stackoverflow.com)

**Comment by Jakobud** [comment]
Okay that makes sense. thanks.
